app/javascript/controllers/media_tag_controller.js

```diff
@@ -73,6 +73,22 @@ export default class extends Controller {
           const event = new CustomEvent('time-update', { detail: timestamp })
           window.dispatchEvent(event)
         })
+       
```

Comment:
Wondering if `loadedmetadata` will reliably give us `textTracks()` if we attach hooks as part of the ready function. Something like:

```javascript
  initializeVideoJSPlayer() {
    this.players = this.mediaTagTargets.map((mediaTag) => {
      mediaTag.classList.add('video-js', 'vjs-default-skin')
      return videojs(mediaTag.id, {}, () => {
        this.on('loadedmetadata', () => {
          const event = new CustomEvent('media-loaded', { detail: player })
          window.dispatchEvent(event)

          // Stop the `loadedmetadata` event and don't bother listening for
          // `timeupdate` events until after `loadedmetadata` fires completes.
          // Why? If we don't, `timeupdate` will be triggered before the video
          // (and its captions) are fully present, and we use the custom
          // `time-update` event to scroll the transcript panel.
          this.off('loadedmetadata')
          this.on('timeupdate', () => {
            const timestamp = this.currentTime()
            const event = new CustomEvent('time-update', { detail: timestamp })
            window.dispatchEvent(event)
          })

          // TODO: Check what's up with this.textTracks() now? Is it reliably present?
          // If so, make sure there are tracks and emit the custom event?
        })
      })
    })
  }
```

Reply:
It may make no difference at all...

Reply:
I tried out several approaches to understand the timing of various events better.  I will add these as comments below here as well on the original issue.  

The summary is the "ready" method passed to the videojs constructor (i.e. player = videojs(mediaTag.id, {}, () => { ...}) is actually called a little before the event handler attached to the "ready" event (i.e. player.on('ready')).  Since "on ready" still doesn't return the cues for the tracks, and that only seems to happen after a given track has been loaded, neither of these are the appropriate time to  trigger transcript loading.